stingray/covariancespectrum.py

```diff
@@ -0,0 +1,428 @@
+            A tuple with minimum and maximum values of the range in the band
+            of interest in reference channel.
+
+        std : float or np.array or list of numbers
+            The term std is used to cacluate the excess variance of a band.
+            If std is set to None, default Poisson case is taken and the
+            std is calculated as `mean(lc)**0.5`. In the case of a single
+            float as input, the same is used as the standard deviation which
+            is also used as the std. And if the std is an iterable of
+            numbers, their mean is used for the same purpose.
+
+
+        Attributes
+        ----------
+        energy_events : dictionary
+            A dictionary with energy bins as keys and time of arrivals of
+            photons with the same energy as value.
+
+        energy_covar : dictionary
+            A dictionary with mid point of band_interest and their covariance
+            computed with their individual reference band. The covaraince
+            values are normalized.
+
+        unnorm_covar : np.ndarray
+            An array of arrays with mid point band_interest and their
+            covariance. It is the array-form of the dictionary `energy_covar`.
+            The covariance values are unnormalized.
+
+        covar : np.ndarray
+            Normalized covaraiance spectrum.
+
+        covar_error : np.ndarray
+            Errors of the normalized covariance spectrum.
+
+        min_time : int
+            Time of arrival of the earliest photon.
+
+        max_time : int
+            Time of arrival of the last photon.
+
+        min_energy : float
+            Energy of the photon with the minimum energy.
+
+        max_energy : float
+            Energy of the photon with the maximum energy.
+
+        Reference
+        ---------
+        [1] Wilkinson, T. and Uttley, P. (2009), Accretion disc variability
+            in the hard state of black hole X-ray binaries. Monthly Notices
+            of the Royal Astronomical Society, 397: 666–676.
+            doi: 10.1111/j.1365-2966.2009.15008.x
+        """
+
+        self.avg_covar = False  # This parameter is used for the purpose of
+                                # AveragedCovariancespectrum.
+
+        self._init_vars(event_list, dt, band_interest,
+                        ref_band_interest, std)
+
+        # A dictionary with energy bin as key and events as value of the key
+        self.energy_events = {}
+
+        self._construct_energy_events(self.energy_events)
+
+        self._update_energy_events(self.energy_events)
+
+        # The dictionary with covariance spectrum for each energy bin
+        self.energy_covar = {}
+
+        self._construct_energy_covar(self.energy_events, self.energy_covar)
+
+    def _init_vars(self, event_list, dt, band_interest,
+                   ref_band_interest, std):
+        if np.all(np.diff(event_list, axis=0).T[0] >= 0) == False:
```

Comment:
This comparison would probably be better with `if not` rather than `if XYZ == False`. Using `== False` or `is False` can lead to surprising results, because `numpy` has its own booleans.